pkg/spoke/helper/helper.go

```diff
@@ -0,0 +1,168 @@
+package helper
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	workapiv1 "open-cluster-management.io/api/work/v1"
+	"open-cluster-management.io/work/pkg/helper"
+	"open-cluster-management.io/work/pkg/spoke/auth"
+)
+
+// DeleteAppliedResources deletes all given applied resources and returns those pending for finalization
+// If the uid recorded in resources is different from what we get by client, ignore the deletion.
+func DeleteAppliedResources(
+	ctx context.Context,
+	resources []workapiv1.AppliedManifestResourceMeta,
+	reason string,
+	dynamicClient dynamic.Interface,
+	recorder events.Recorder,
+	owner metav1.OwnerReference,
+	kubeClient kubernetes.Interface, executor *workapiv1.ManifestWorkExecutor) ([]workapiv1.AppliedManifestResourceMeta, []error) {
```

Comment:
I think you want to input validator here, we need to consider cache in the future.